lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1345,6 +1345,64 @@ static void optimizeBatchNorm(Function *F) {
   } // For all nodes in the graph.
 }
 
+/// Return true if the \p node sub-tree corresponds to a scalar
+/// Constant node and return the float value in \p retFloat.
+static bool getFloatScalarConstant(Node *node, float *retFloat) {
+  // Iterate accross potential Tile Nodes (implied by broadcasting if any).
+  auto *n = node;
+  while (auto *TN = dyn_cast<TileNode>(n)) {
+    n = TN->getInput();
+  }
+
+  // After potential Tile nodes, it should a singleton Constant node with any
+  // shape corresponding to one single element.
+  if (Constant *constNode = dyn_cast<Constant>(n)) {
```

Comment:
Please do early return:
```suggestion
  Constant *constNode = dyn_cast<Constant>(n);
  if (!constNode || constNode->getType()->getElementType() != ElemKind::FloatTy
      || constNode->getType()->size() != 1)
    return false;
  ...
```

Reply:
Done